clash-prelude/src/Clash/Explicit/BlockRam/File.hs

```diff
@@ -205,6 +221,79 @@ blockRamFile = \clk gen sz file rd wrM ->
       (blockRamFile# clk gen sz file (fromEnum <$> rd) en (fromEnum <$> wr) din)
 {-# INLINE blockRamFile #-}
 
+-- | Convert data to the String contents of a memory file.
+--
+-- * __NB:__ Not synthesizable
+-- * The following document the several ways to instantiate components with
+-- files:
+--
+--     * "Clash.Prelude.BlockRam.File#usingramfiles"
+--     * "Clash.Prelude.ROM.File#usingromfiles"
+--     * "Clash.Explicit.BlockRam.File#usingramfiles"
+--     * "Clash.Explicit.ROM.File#usingromfiles"
+--
+-- * See "Clash.Sized.Fixed#creatingdatafiles" for more ideas on how to create
+-- your own data files.
+--
+-- = Example
+--
+-- The @Maybe@ datatype has don't care bits, where the actual value does not
+-- matter. But the bits need a defined value in the memory. Either 0 or 1 can be
+-- used, and both are valid representations of the data.
+--
+-- >>> let es = [ Nothing, Just (7 :: Unsigned 8), Just 8]
+-- >>> mapM_ (putStrLn . show . pack) es
+-- 0b0_...._....
+-- 0b1_0000_0111
+-- 0b1_0000_1000
+-- >>> putStr (memFile (Just 0) es)
+-- 000000000
+-- 100000111
+-- 100001000
+-- >>> putStr (memFile (Just 1) es)
+-- 011111111
+-- 100000111
+-- 100001000
+--
+memFile
+  :: forall a
+   . ( BitPack a
+     , HasCallStack)
+  => Maybe Bit
+  -- ^ Value to map don't care bits to. Nothing means throwing an error on
+  -- don't care bits.
+  -> [a]
```

Comment:
I wonder if people may want to convert a `Vec` of values? Maybe `[a]` should be `(Foldable f) => f a`?

Reply:
I wrote this function because `Vec`s were impossibly slow and memory hungry for me :grin:. But I could do `Foldable` if you prefer. Shall I change it to that?

Reply:
Done